Upstream-Status: Backport

From 6f64495840c4e5674d542ccf20df96ed12665687 Mon Sep 17 00:00:00 2001
From: Markus Vieth <mvieth@techfak.uni-bielefeld.de>
Date: Tue, 5 Nov 2024 20:07:13 +0100
Subject: [PATCH] Prepare for Boost 1.87 Some stuff from Boost's asio library
 has been removed after Boost 1.86. The documentation says that it has been
 deprecated, but no compiler warnings are shown. This page explains what is
 considered "old" and what should be used instead: 
 https://www.boost.org/doc/libs/1_86_0/doc/html/boost_asio/net_ts.html -
 io_service was simply an alias (typedef) for io_context - make_address
 replaces from_string - resolver.resolve and connect work a bit different now,
 see also
 https://www.boost.org/doc/libs/1_86_0/doc/html/boost_asio/tutorial/tutdaytime1/src.html

---
 apps/src/openni_mobile_server.cpp         | 2 +-
 apps/src/openni_octree_compression.cpp    | 4 ++--
 apps/src/openni_organized_compression.cpp | 4 ++--
 io/include/pcl/io/hdl_grabber.h           | 2 +-
 io/include/pcl/io/robot_eye_grabber.h     | 2 +-
 io/include/pcl/io/tim_grabber.h           | 2 +-
 io/src/hdl_grabber.cpp                    | 2 +-
 io/src/robot_eye_grabber.cpp              | 2 +-
 io/src/tim_grabber.cpp                    | 4 ++--
 io/src/vlp_grabber.cpp                    | 2 +-
 10 files changed, 13 insertions(+), 13 deletions(-)

Index: git/apps/src/openni_mobile_server.cpp
===================================================================
--- git.orig/apps/src/openni_mobile_server.cpp
+++ git/apps/src/openni_mobile_server.cpp
@@ -157,7 +157,7 @@ public:
 
     viewer_.showCloud(getLatestPointCloud());
 
-    boost::asio::io_service io_service;
+    boost::asio::io_context io_service;
     tcp::endpoint endpoint(tcp::v4(), static_cast<unsigned short>(port_));
     tcp::acceptor acceptor(io_service, endpoint);
     tcp::socket socket(io_service);
Index: git/apps/src/openni_octree_compression.cpp
===================================================================
--- git.orig/apps/src/openni_octree_compression.cpp
+++ git/apps/src/openni_octree_compression.cpp
@@ -415,7 +415,7 @@ main(int argc, char** argv)
     if (bEnDecode) {
       // ENCODING
       try {
-        boost::asio::io_service io_service;
+        boost::asio::io_context io_service;
         tcp::endpoint endpoint(tcp::v4(), 6666);
         tcp::acceptor acceptor(io_service, endpoint);
 
@@ -423,7 +423,7 @@ main(int argc, char** argv)
 
         std::cout << "Waiting for connection.." << std::endl;
 
-        acceptor.accept(*socketStream.rdbuf());
+        acceptor.accept(socketStream.rdbuf()->socket());
 
         std::cout << "Connected!" << std::endl;
 
Index: git/apps/src/openni_organized_compression.cpp
===================================================================
--- git.orig/apps/src/openni_organized_compression.cpp
+++ git/apps/src/openni_organized_compression.cpp
@@ -438,7 +438,7 @@ main(int argc, char** argv)
     if (bEnDecode) {
       // ENCODING
       try {
-        boost::asio::io_service io_service;
+        boost::asio::io_context io_service;
         tcp::endpoint endpoint(tcp::v4(), 6666);
         tcp::acceptor acceptor(io_service, endpoint);
 
@@ -446,7 +446,7 @@ main(int argc, char** argv)
 
         std::cout << "Waiting for connection.." << std::endl;
 
-        acceptor.accept(*socketStream.rdbuf());
+        acceptor.accept(socketStream.rdbuf()->socket());
 
         std::cout << "Connected!" << std::endl;
 
Index: git/io/include/pcl/io/hdl_grabber.h
===================================================================
--- git.orig/io/include/pcl/io/hdl_grabber.h
+++ git/io/include/pcl/io/hdl_grabber.h
@@ -274,7 +274,7 @@ namespace pcl
       boost::asio::ip::udp::endpoint udp_listener_endpoint_;
       boost::asio::ip::address source_address_filter_;
       std::uint16_t source_port_filter_;
-      boost::asio::io_service hdl_read_socket_service_;
+      boost::asio::io_context hdl_read_socket_service_;
       boost::asio::ip::udp::socket *hdl_read_socket_;
       std::string pcap_file_name_;
       std::thread *queue_consumer_thread_;
Index: git/io/include/pcl/io/robot_eye_grabber.h
===================================================================
--- git.orig/io/include/pcl/io/robot_eye_grabber.h
+++ git/io/include/pcl/io/robot_eye_grabber.h
@@ -131,7 +131,7 @@ namespace pcl
 
       boost::asio::ip::address sensor_address_;
       boost::asio::ip::udp::endpoint sender_endpoint_;
-      boost::asio::io_service io_service_;
+      boost::asio::io_context io_service_;
       std::shared_ptr<boost::asio::ip::udp::socket> socket_;
       std::shared_ptr<std::thread> socket_thread_;
       std::shared_ptr<std::thread> consumer_thread_;
Index: git/io/include/pcl/io/tim_grabber.h
===================================================================
--- git.orig/io/include/pcl/io/tim_grabber.h
+++ git/io/include/pcl/io/tim_grabber.h
@@ -128,7 +128,7 @@ class PCL_EXPORTS TimGrabber : public Gr
     std::vector<float> distances_;
 
     boost::asio::ip::tcp::endpoint tcp_endpoint_;
-    boost::asio::io_service tim_io_service_;
+    boost::asio::io_context tim_io_service_;
     boost::asio::ip::tcp::socket tim_socket_;
     //// wait time for receiving data (on the order of milliseconds)
     unsigned int wait_time_milliseconds_ = 0;
Index: git/io/src/hdl_grabber.cpp
===================================================================
--- git.orig/io/src/hdl_grabber.cpp
+++ git/io/src/hdl_grabber.cpp
@@ -287,7 +287,7 @@ pcl::HDLGrabber::loadHDL32Corrections ()
 boost::asio::ip::address
 pcl::HDLGrabber::getDefaultNetworkAddress ()
 {
-  return (boost::asio::ip::address::from_string ("192.168.3.255"));
+  return (boost::asio::ip::make_address ("192.168.3.255"));
 }
 
 /////////////////////////////////////////////////////////////////////////////
Index: git/io/src/robot_eye_grabber.cpp
===================================================================
--- git.orig/io/src/robot_eye_grabber.cpp
+++ git/io/src/robot_eye_grabber.cpp
@@ -269,7 +269,7 @@ void
 pcl::RobotEyeGrabber::socketThreadLoop ()
 {
   asyncSocketReceive();
-  io_service_.reset();
+  io_service_.restart();
   io_service_.run();
 }
 
Index: git/io/src/tim_grabber.cpp
===================================================================
--- git.orig/io/src/tim_grabber.cpp
+++ git/io/src/tim_grabber.cpp
@@ -184,8 +184,8 @@ pcl::TimGrabber::start ()
 
   try {
     boost::asio::ip::tcp::resolver resolver (tim_io_service_);
-    tcp_endpoint_ = *resolver.resolve (tcp_endpoint_);
-    tim_socket_.connect (tcp_endpoint_);
+    boost::asio::ip::tcp::resolver::results_type endpoints = resolver.resolve (tcp_endpoint_);
+    boost::asio::connect(tim_socket_, endpoints);
   }
   catch (std::exception& e)
   {
Index: git/io/src/vlp_grabber.cpp
===================================================================
--- git.orig/io/src/vlp_grabber.cpp
+++ git/io/src/vlp_grabber.cpp
@@ -92,7 +92,7 @@ pcl::VLPGrabber::loadVLP16Corrections ()
 boost::asio::ip::address
 pcl::VLPGrabber::getDefaultNetworkAddress ()
 {
-  return (boost::asio::ip::address::from_string ("255.255.255.255"));
+  return (boost::asio::ip::make_address ("255.255.255.255"));
 }
 
 /////////////////////////////////////////////////////////////////////////////
